internal/character/gepard/eidolon.go

```diff
@@ -0,0 +1,79 @@
+package gepard
+
+import (
+	"github.com/simimpact/srsim/internal/global/common"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+)
+
+const (
+	E2    key.Modifier = "gepard-e2"
+	E4    key.Modifier = "gepard-e4"
+	E4Res key.Modifier = "gepard-e4-res"
+)
+
+func init() {
+	modifier.Register(E4, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnAdd: func(mod *modifier.ModifierInstance) {
+				targets := mod.Engine().Characters()
+
+				for _, trg := range targets {
+					mod.Engine().AddModifier(trg, info.Modifier{
+						Name:   E4Res,
+						Source: mod.Owner(),
+					})
+				}
+
+				mod.Engine().Events().CharacterAdded.Subscribe(func(e event.CharacterAddedEvent) {
+					mod.Engine().AddModifier(e.Id, info.Modifier{
+						Name:   E4Res,
+						Source: mod.Owner(),
+					})
+				})
+			},
+			OnBeforeDying: func(mod *modifier.ModifierInstance) {
+				targets := mod.Engine().Characters()
+
+				for _, trg := range targets {
+					mod.Engine().RemoveModifier(trg, E4Res)
+				}
+			},
+		},
+	})
```

Comment:
IMO this logic modifier is not necessary. Instead I would:
* Move all of the `OnAdd` logic to the `e4()` init function
* On the `E4` modifier that is applying res (should just be a singular `gepard-e4` modifier that applies to everyone), give this modifier the `OnBeforeDying` listener. In that modifier, check if the owner is Gepard (gepard is dying). If so, remove the modifier from all allies